avro-fastserde/src/main/java/com/linkedin/avro/fastserde/SchemaAssistant.java

```diff
@@ -68,6 +68,15 @@ public static boolean isComplexType(Schema schema) {
     }
   }
 
+  public static String getSchemaFullName(Schema schema) {
+    Schema.Type type = schema.getType();
+    boolean isNamedType = type.equals(Schema.Type.ENUM) || type.equals(Schema.Type.FIXED) || type.equals(Schema.Type.RECORD);
+    /**
+     * Avro-1.4 doesn't support {@link Schema#getFullName()} if the Schema is not a NamedSchema.
```

Comment:
EnumSchema extends NamedSchema even in 1.4

Reply:
Yes, that is why it is allowed in the code path of Avro-1.4.